### PR TITLE
[ ResNet ] Add source code to run ResNet

### PR DESCRIPTION
**Pull request by jijoongmoon**
This PR includes.
  jni/main.cpp : run resnet
  gen_input : generate cifar10 input
  res/resnet.ini : update ini file

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: jijoong.moon <jijoong.moon@samsung.com>

### CODE REVIEW

File: Applications/ResNet/jni/main.cpp [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+ * @file	main.cpp
+ * @date	27 Jan 2021
+ * @see		https://github.com/nnstreamer/nntrainer
+ * @author	Jijoong Moon <jijoong.moon@samsung.com>
+ * @bug		No known bugs except for NYI items
+ * @brief	This is Resnet18 Example
+ *
+ */
+
+#include "bitmap_helpers.h"
```
Comment (kparichay) [inline Applications/ResNet/jni/main.cpp:14]: #include "bitmap_helpers.h" -> #include <bitmap_helpers.h>

File: Applications/ResNet/jni/main.cpp [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+#include <stdlib.h>
+
+#include "databuffer.h"
```
Comment (kparichay) [inline Applications/ResNet/jni/main.cpp:21]: `#include "" -> <>` for all the nntrainer includes

File: Applications/ResNet/jni/main.cpp [hunk 3]
```diff
@@ -0,0 +1,421 @@
...
+    return 0;
+  }
+
+  try {
+    NN.readModel();
+    NN.setDataBuffer((DB));
+    NN.train();
+    training_loss = NN.getTrainingLoss();
+    validation_loss = NN.getValidationLoss();
+    last_batch_loss = NN.getLoss();
```
Comment (kparichay) [inline Applications/ResNet/jni/main.cpp:414]: can you add the results of this training in the PR description?

File: Applications/ResNet/res/resnet18.ini [hunk 4]
```diff
@@ -154,12 +154,20 @@ kernel_size = 3,3
 bias_initializer = zeros
 weight_initializer = xavier_uniform
 filters=128
-stride =2,2
+stride =1,1
 padding=1,1
 
+[B2_pooling2d_1]
```
Comment (kparichay) [inline Applications/ResNet/res/resnet18.ini:160]: How about instead of modifying the original resnet18.ini, we keep a resnet18-pooling.ini, so that resnet18.ini can be tested once @zhoonit resolves the multi-stride?

File: Applications/ResNet/gen_input/convert.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        path = path + (label_name[class_id])+'/'+ '{:04d}'.format(500-(index % num_val))+'.bmp'
+    else:
+        class_id = index // num_train
+        path = path + (label_name[class_id])+'/'+ '{:04d}'.format((index % num_train)+1)+'.bmp'
+
+    image = Image.open(path)
+    data = np.array(image)
+    data = np.transpose(data, [2,0,1])
+
+    label = np.zeros(num_class, dtype=np.uint8)
```
Comment (lhs8928) [inline Applications/ResNet/gen_input/convert.py:53]: Simple question:
It is okay to use np.uint8 for dtype?
Cause getData function in main.cpp the size of num_class is treated as float but the range of np.uint8 is 0 to 255

File: Applications/ResNet/jni/main.cpp [hunk 6]
```diff
@@ -0,0 +1,421 @@
...
+std::string resource;
+
+float training_loss = 0.0;
+float validation_loss = 0.0;
+float last_batch_loss = 0.0;
+
+typedef struct {
+  unsigned int remain;
+  std::vector<unsigned int> duplication;
+} counting_info_s;
```
Comment (lhs8928) [inline Applications/ResNet/jni/main.cpp:61]: Instead of duplication vector shuffle the index vector might also a alternative way to handle the random data I guess :)

Like

vector<int> index;
std::iota(index.begin(), index.end(), 0);
std::shuffle(index.begin(), index.end(), rand_engine);

This way can prevent erase, clear, resize the vector

File: meson.build [hunk 7]
```diff
@@ -39,6 +39,8 @@ warning_flags = [
   '-Wno-error=varargs',
   '-O2',
   '-ftree-vectorize'
+#  '-g'
```
Comment (zhoonit) [inline meson.build:42]: I wonder if this is intended

File: Applications/ResNet/jni/meson.build [hunk 8]
```diff
@@ -0,0 +1,18 @@
+build_root = meson.build_root()
+res_path = meson.current_source_dir() / '..' / 'res'
+mnist_sources = [
+  'main.cpp'
+]
+if build_platform == 'tizen'
```
Comment (zhoonit) [inline Applications/ResNet/jni/meson.build:6]: I guess this check can be deleted?

**Comment by zhoonit** [comment]
I wonder if we could fix and merge this :)

**Comment by jijoongmoon** [comment]
This PR includes related PR #1339 1336
